Add docs about versioning (#534)
cytomining/pycytominer: CONTRIBUTING.md

### Releases

Project maintainers are responsible for releasing new versions of pycytominer.
We generally abide [PEP 440](https://packaging.python.org/en/latest/specifications/version-specifiers/#version-specifiers) for version specifications and use [`poetry-dynamic-versioning`](https://github.com/mtkennerly/poetry-dynamic-versioning) to help facilitate this work.

> Docker-image focused versioning has a unique specification which builds upon and differs from our package versioning (please see the [relevant section below](#docker-hub-image-releases)).

Reviewer: ```suggestion
Note: we use a slightly different specification for versioning our Docker image (please see the [relevant section below](#docker-hub-image-releases)).
```

Not sure I love my suggestion, but I think it improves clarity a bit
